tools/tpm2_createprimary.c

```diff
@@ -293,24 +294,26 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
     return *opts != NULL;
 }
 
-int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
+#define VALID_CTX(ctx) \
+	(ctx.flags.A && ctx.flags.g && ctx.flags.G)
 
+int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
     UNUSED(flags);
-    int returnVal = 0;
+    int returnVal;
 
-    if(ctx.flags.A == 1 && ctx.flags.g == 1 && ctx.flags.G == 1) {
-        returnVal = create_primary(sapi_context);
+    if (VALID_CTX(ctx) != true) {
+        LOG_ERR("Invalid option flags");
+        return -EINVAL;
+    }
 
-        if (returnVal == 0 && ctx.flags.C) {
-            returnVal = files_save_tpm_context_to_path(sapi_context, ctx.handle2048rsa,
-                                                       ctx.context_file) != true;
-        }
+    returnVal = create_primary(sapi_context);
+    if (returnVal != 0 || ctx.flags.C != 1) {
```

Comment:
```if (!returnVal || !ctx.flags.C)```

If you wanted to kill flags, you could check if ctx.context_file is set.

Reply:
@williamcroberts in this case change_primary() doesn't just return 0 or 1 so you can't do !returnVal. This is why in general I feel it's better to do an explicit check against value if returning an int rather than a bool.